src/tree_schema_helpers.c

```diff
@@ -1471,50 +1380,20 @@ lysc_node_children(const struct lysc_node *node, uint16_t flags)
         return NULL;
     }
 
-    children = lysc_node_children_p(node, flags);
-    if (children) {
-        return *children;
-    } else {
-        return NULL;
-    }
-}
-
-API const struct lysc_node *
-lysc_node_children_full(const struct lysc_node *node, uint16_t flags)
-{
-    if (!node) {
-        return NULL;
-    }
-
-    if (node->nodetype == LYS_RPC || node->nodetype == LYS_ACTION) {
+    if ((node->nodetype == LYS_RPC) || (node->nodetype == LYS_ACTION)) {
         if (flags & LYS_CONFIG_R) {
-            return (struct lysc_node *)&((struct lysc_action *)node)->output;
+            return (struct lysc_node *)&((struct lysc_node_action *)node)->output;
         } else {
             /* LYS_CONFIG_W, but also the default case */
-            return (struct lysc_node *)&((struct lysc_action *)node)->input;
+            return (struct lysc_node *)&((struct lysc_node_action *)node)->input;
         }
```

Comment:
Also I suggest creating type conversion inline functions to avoid these casts in multiple places. This allows dropping in an assert as well e.g.,
```
static inline struct lysc_node_action *
lysc_node_action(struct lysc_node *n)
{
    assert(n->nodetype == LYS_ACTION);
    return (struct lysc_node_action *)n;
}
```
then the above code becomes:
`return &lysc_node_action(node)->input.node;`
which eliminates all casting.

Reply:
good point, I like it